### ISSUE TITLE
Feature request : Directional swap

### ISSUE THREAD

**Issue by DonaldoDes** [ianyh/Amethyst#1305]
Hi folks !

There are currently two ways to reorder windows on a partitioned layout like "Tall" or "BSP", the first one is through cycling all the windows and the other is through swapping focused window with the main window

Like for the directional focus feature requested in #521 it would be really useful to have a directional swap feature

**Comment by dyllandry** [comment]
Yeah I think I'd need this feature to use focus swapping in Amethyst. Coming from Vim and Tmux, directional swapping is all I've ever known. Visualizing a clock to know how to get to the window I want feels cumbersome when I'm used to thinking "go left" or "go up".
